### PR TITLE
Write integration tests

### PR DESCRIPTION
**Pull request by emmiiorji**
This change adds test for the index and show views of posts and users.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class PostsController < ApplicationController
   def index
     @users = User.all
     @user = User.find_by(id: params[:user_id])
-    @posts = @user ? Post.where(author: @user) : nil
+    @posts = @user ? Post.where(author: @user).paginate(page: params[:page], per_page: 5) : nil
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb:5]: - [ ] To solve the N+1 issue to be able to fetch all posts and their comments for a user you can simply do:
`@posts = @user.posts.includes(:comments)`

File: spec/features/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    button = find("#{@posts_url} button")
+    expect(button).to have_text('See all posts')
+  end
+
+  it 'redirects to posts index on clicking of the See all posts button' do
+    find(@posts_url).click
+    new_post_anchor = find("a[href='#{new_user_post_path(@user.id)}']")
+    expect(new_post_anchor).to have_content('New Post')
+  end
+end
```
Comment (DeliceLydia) [inline spec/features/users/show.html.erb_spec.rb]: - [ ] Please fix one test that is failing
Screenshot from 2022-11-03 17-55-22

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @emmiiorji, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


### Cheers!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (emmiiorji) [reply app/controllers/posts_controller.rb:5]: Hi @DeliceLydia,

Thanks for the kind review.

We don't seem to get a warning from bullet about it. We have the [bullet gem](https://github.com/flyerhzm/bullet) installed which checks for n+1 queries. We've solved the issues it identified as n+1 queries.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @emmiiorji and @HansZizold, 

Good job implementing the changes requested by the previous reviewer.
To highlight 👍🏽 
- You wrote integration tests
- Your tests pass

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**